src/App.tsx

```diff
@@ -1,14 +1,42 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [typeOfTodos, setTypeOfTodos] = useState('all');
+  const [filter, setFilter] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [todoToShow, setTodoToShow] = useState<Todo | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      setIsLoading(true);
+      const todosFromServer = await getTodos();
+
+      setTodos(todosFromServer);
+      setIsLoading(false);
+    };
+
+    fetchTodos();
+  }, []);
+
+  const groupedTodos = typeOfTodos === 'all'
+    ? todos
+    : todos.filter(({ completed }) => (typeOfTodos === 'completed' ? completed : !completed));
+
+  const filteredTodos = filter === ''
+    ? groupedTodos
+    : groupedTodos.filter(({ title }) => title.toLowerCase().includes(filter.toLowerCase()));
```

Comment:
create separate function for this logic and use switch case for typeOfTodos
Also, move filter.toLowerCase() to a separate variable, don't do it on each filter iteration